### PR TITLE
Fix case where chained fixups has less segments

### PR DESCRIPTION
**Pull request by akihikodaki**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why. Reason: no documentation present.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible) Reason: the binary is not redistributable.
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed) Reason: not needed.

**Detailed description**

If you install Kernel Debug Kit 12.2 build 21D49, you'll get:
`/Library/Developer/KDKs/KDK_12.2_21D49.kdk/System/Library/Kernels/kernel.development.t8101`

The file has 23 segments, but contains only 22 dyld chained fixups.

**Test plan**

1. Install Kernel Debug Kit 12.2 build 21D49. For details, see: https://developer.apple.com/documentation/apple-silicon/debugging-a-custom-kernel-extension
2. Open `/Library/Developer/KDKs/KDK_12.2_21D49.kdk/System/Library/Kernels/kernel.development.t8101` and confirm it does not crash.

**Closing issues**

### CODE REVIEW

**Comment by thestr4ng3r** [comment]
The approach overall looks good to me, but there still seems to be something off making some tests fail, e.g. https://github.com/rizinorg/rizin/runs/5494437747?check_suite_focus=true#step:16:517
I can also check in detail tonight.

(By the way, thanks a lot for your guide and patches at https://gist.github.com/akihikodaki/87df4149e7ca87f18dc56807ec5a1bc5. It has been quite useful to me 😊)

**Comment by akihikodaki** [comment]
@thestr4ng3r b3fbbafbf8c36e253b5caa1baacb300a6c31e964: Amended the fixup reconstruction code, hopefully fixing the CI failure.

(I didn't expect to see someone who uses my QEMU fork here. I wrote it for my personal use, but I'm glad to know you find it useful.)

File: librz/bin/format/mach0/mach0.c [hunk 1]
```diff
@@ -1537,6 +1536,7 @@ static bool reconstruct_chained_fixup(struct MACH0_(obj_t) * bin) {
 	if (!bin->nsegs) {
 		return false;
 	}
+	bin->nchained_starts = bin->nsegs;
 	bin->chained_starts = RZ_NEWS0(struct rz_dyld_chained_starts_in_segment *, bin->nsegs);
```
Comment (thestr4ng3r) [inline librz/bin/format/mach0/mach0.c]:
```suggestion
	bin->chained_starts = RZ_NEWS0(struct rz_dyld_chained_starts_in_segment *, bin->nchained_starts);
```
Just a small change to express the meaning of `nchained_starts` better.

**Review by thestr4ng3r — APPROVED** [review]
Ok, I see it was because some binaries have the `LC_DYLD_CHAINED_FIXUPS` load command and some use the `LC_DYLD_INFO` instead. Looks good to me now!

**Comment by thestr4ng3r** [comment]
Thanks! If you are interested in macOS arm64 support in general, feel free to also join our mattermost: https://rizin.re/community We are trying to support these things as good as possible in our given time and resource constraints.

**Pull request merged**